### PR TITLE
Log the exact version at init

### PR DESCRIPTION
**Pull request by NiLuJe**
Might come in handy one day, because I know quite a few people extremely averse to updating stuff ^^.

e.g.,
```
May 13 17:57:13 nickel: (NickelMenu) version: v0.0.3-2-g043dfb3 (src/init.c:24)
May 13 17:57:13 nickel: (NickelMenu) init: creating failsafe (src/init.c:25)
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +39,12 @@ LDFLAGS  ?= -Wl,--as-needed
 override CFLAGS   += -std=gnu11 -Wall -Wextra -Werror
 override CXXFLAGS += -std=gnu++11 -Wall -Wextra -Werror
 override LDFLAGS  += -Wl,--no-undefined -Wl,-rpath,/usr/local/Kobo -Wl,-rpath,/usr/local/Qt-5.2.1-arm/lib
+
+NM_VERSION:=$(shell git describe --tags)
```
Comment (geek1011) [inline Makefile:43]:
```suggestion
NM_VERSION := $(shell git describe --tags --always --dirty)
```

File: src/util.h [hunk 2]
```diff
@@ -13,6 +13,11 @@ extern "C" {
 #include <string.h>
 #include <syslog.h>
 
+// Fallback version tag
+#ifndef NM_VERSION
+#   define NM_VERSION "dev"
```
Comment (geek1011) [inline src/util.h:18]:
```suggestion
#define NM_VERSION "dev"
```

Just to keep things consistent.

**Comment by geek1011** [comment]
^ A note about branch deletion and squash/merge: I've settled on using squash commits and deleting branches for small changes which are tracked in a PR. For larger changes, I'll use rebasing unless a merge commit would be more suitable.

**Comment by NiLuJe** [comment]
:+1:

(You'll notice I've been reusing a "disposable" topical branch named "misc" for trivial stuff. I'll keep meatier features in named branches on my end).

**Comment by NiLuJe** [comment]
@geek1011: Complete sidebar: I'll be including NM in the [KOReader/Plato OCP packages](https://www.mobileread.com/forums/showthread.php?t=314220), unless you have an objection to that ;).

**Comment by geek1011** [comment]
@NiLuJe: Sounds good. One question: how are you planning to deal with updates?

**Comment by NiLuJe** [comment]
I try to rebuild those ASAP on any new Plato/KOReader release, and/or any significant KFMon changes (i.e., it's actually always picking up a KFMon git build).

So, I'll just add NM to the list of stuff that might require a rebuild ;).

**Comment by geek1011** [comment]
OK.

> EDIT: In practice, that means either using the latest release via the GitHub API, or using a local build like for KFMon.

Note that you can link directly to the latest release with https://github.com/geek1011/NickelMenu/releases/latest/download/KoboRoot.tgz.

**Pull request merged**